forte/data/data_store.py

```diff
@@ -198,14 +202,161 @@ def __init__(
+            2) recreate the `entry_dict` from `__elements`;
+        """
+        super().__setstate__(state)
+        self.__elements = self.__dict__.pop("entries")
+        # add `_type_attributes` to this object to check attributes
+        self._type_attributes = self.__dict__.pop("fields", {})
+        # recreate the `entry_dict` from `__elements`
+        self._DataStore__entry_dict = {}
+
+        for k in self.__elements:
+            if self._is_annotation(k):
+                self.__elements[k] = SortedList(self.__elements[k])
+            for e in self.__elements[k]:
+                # retrieve tid for every entry
+                self._DataStore__entry_dict[e[2]] = e
+
+    @classmethod
+    def deserialize(
+        cls,
+        data_source: str,
+        serialize_method: str = "json",
+        check_attribute: bool = True,
+        silent_mode: bool = True,
+        accept_none: bool = True,
+    ) -> "DataStore":
+        """
+        Deserialize a data_store from a string. This internally calls the
+        internal :meth:`~forte.data.base_store.BaseStore._deserialize` function
+        from :class:`~forte.data.base_store.BaseStore`.
+
+        Args:
+            data_source: The path storing data source.
+            serialize_method: The method used to serialize the data, this
+                should be the same as how serialization is done. The current
+                option is `json`.
+            check_attribute: Boolean value indicating whether users want to
+                check compatibility of attributes. Only applicable when
+                `save_attribute` is set to True in BaseStore.serialize.
+            silent_mode: Boolean value indicating whether users want to
+                see warnings when it checks attributes.
+            accept_none: Boolean value indicating whether users want to
+                fill contradictory fields with none. Only applicable when
+                `check_attribute` is set to True. If false, it will raise
+                an ValueError if there are any contradictions in fields.
+
+        Returns:
+            An data store object deserialized from the string.
+        """
+        store = cls._deserialize(data_source, serialize_method)
+        # The following part is for customized json method only.
+        if isinstance(store, dict):
+            obj = DataStore()
+            obj.__setstate__(store)
+            store = obj
+        if check_attribute:
+            if len(store._type_attributes) == 0:
+                raise ValueError(
+                    "Saved object does not support check_attribute."
+                )
+            if cls._type_attributes != store._type_attributes:
+                # find the `entry_type` with different fields, count different
+                # fields. `diff` records fields in the current class but not
+                # in the serialized objects.
+                for t, v in cls._type_attributes.items():
```

Comment:
better to put this whole code block in another function

Reply:
fixed